swayidle/main.c

```diff
@@ -368,7 +369,7 @@ static int display_event(int fd, uint32_t mask, void *data) {
 	if (mask & WL_EVENT_HANGUP) {
 		sway_terminate(0);
 	}
-	if (wl_display_dispatch(state.display) < 0) {
+	if (wl_display_dispatch_pending(state.display) < 0) {
```

Comment:
I don't think this works either. `wl_display_dispatch_pending`'s description says:

> Dispatch default queue events without reading from the display fd. 

I think we need to use `wl_display_read_events`. See how mako does it:

https://github.com/emersion/mako/blob/master/event-loop.c#L238